x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState } from 'react';
+import { Direction, EuiBasicTableProps, EuiTableSortingType } from '@elastic/eui';
+import sortBy from 'lodash/sortBy';
+import { DataFrameAnalyticsListColumn, DataFrameAnalyticsListRow } from './common';
+
+const PAGE_SIZE = 10;
+const PAGE_SIZE_OPTIONS = [10, 25, 50];
+
+interface AnalyticsBasicTableSettings {
+  pageIndex: number;
+  pageSize: number;
+  totalItemCount: number;
+  hidePerPageOptions: boolean;
+  sortField: string;
+  sortDirection: Direction;
+}
+
+interface UseTableSettingsReturnValue {
+  onTableChange: EuiBasicTableProps<DataFrameAnalyticsListRow>['onChange'];
+  pageOfItems: DataFrameAnalyticsListRow[];
+  pagination: EuiBasicTableProps<DataFrameAnalyticsListRow>['pagination'];
+  sorting: EuiTableSortingType<any>;
+}
+
+export function useTableSettings(items: DataFrameAnalyticsListRow[]): UseTableSettingsReturnValue {
+  const [tableSettings, setTableSettings] = useState<AnalyticsBasicTableSettings>({
+    pageIndex: 0,
+    pageSize: PAGE_SIZE,
+    totalItemCount: 0,
+    hidePerPageOptions: false,
+    sortField: DataFrameAnalyticsListColumn.id,
+    sortDirection: 'asc',
+  });
+
+  const getPageOfItems = (
+    list: any[],
+    index: number,
+    size: number,
+    sortField: string,
+    sortDirection: Direction
+  ) => {
+    list = sortBy(list, (item) => item[sortField]);
```

Comment:
Not sure if this worked before, but sorting by job type doesn't seem to work:

![image](https://user-images.githubusercontent.com/7405507/91330366-390bcb00-e7c1-11ea-8090-b875d1551535.png)

Reply:
Thanks for the catch, @peteharverson  - fixed this in a0350d02770c47d000155f6c1abc841aec7745b3